### ISSUE TITLE
Decouple energy API

### ISSUE THREAD

**Issue by InfiniteCoder01** [mt-mods/technic#388]
Is there any possiblilty to decouple energy API into a separate mod for use with other mods without bringing the whole technic with them?

**Comment by S-S-X** [comment]
Yes but might need some work. But it isn't very good. Actually it's terrible.
However this would allow making it less terrible, decoupling would likely also be better for technic itself in the long run (likely a bit annoying at first but that's expected with changes like this).

Few years ago I did some preliminary work to decouple some stuff from network under the hood but there's still few things to do, these come into mind:
* #108 Some work done so a bit better but still incomplete. Likely should be taken into account, it probably could keep doing what it does but this is one good example for how to not handle things.
* #124 Not exactly for decoupling but related as this might need some generic solution at network API level, not sure. Likely can be fixed afterward too)
* #264 Some work done so a bit better but still incomplete, this kind of has to be done.
* Supply converters might need some work, these are also a good example case for what such (energy) network mod / framework would need.

Previously completed related work, possibly some of the descriptions / change lists might be useful for planning (basically just picked my old pr's for directly network related work):
* #80 
* #96
* #102
* #126
* #136
* #152
* #241
* #265

That said, there _might_ be better alternatives to technic. Though can't name any.
